File: 18_truncate_a_string.js
// truncate a string

function truncateString(str, num){
    if (num >= str.length) return str;
    else if (num <= 3) return str.slice(0, num) + "...";
    return str.slice(0, num-3) + "...";
}


console.log(truncateString("A-tisket a tasket A green and yellow basket", 11));
console.log(truncateString("tommy goes to work.", 3));
console.log(truncateString("A-tisket a tasket A green and yellow basket","A-tisket a tasket A green and yellow basket".length + 2));

File: 18_truncate_a_string.py
# truncate a string

def truncateString(string, num):
    if num >= len(string):
        return string
    elif num <= 3:
        return string[0:num] + "..."
    else:
        return string[0:num-3] + "..."





print(truncateString("A-tisket a tasket A green and yellow basket", 11))
print(truncateString("tommy goes to work.", 3))
print(truncateString("A-tisket a tasket A green and yellow basket",len("A-tisket a tasket A green and yellow basket") + 2))
